### PR TITLE
ci: replace travis completely with github actions

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
- .github/workflows/deploy.yml: set up image builds andreleasing with github actions
- Makefile,release/Makefile: remove multiarch image build targets, since all non-host arch builds can be done in action configs
- *: remove Travis CI config from project and references in docs

**Motivation for the change:** for various reasons (cost, reliability, debuggability, visibility) Travis is no longer viable for the operator-sdk project. Github Actions, via `docker buildx`, can support cross-platform image builds, release automation, and is _way_ more configurable/ergonomic.

This work is based off of https://github.com/joelanford/helm-operator/blob/7fe2dd3/.github/workflows/deploy.yml.

/area testing

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [x] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by estroz** [comment]
Passing run of this PR: https://github.com/estroz/operator-sdk/runs/1916369513?check_suite_focus=true

I still need to add the proper secret environment variables for this to work, and plan on doing so tomorrow.

/cc @joelanford

**Comment by estroz** [comment]
As a follow-up, I'd like to start nightly builds of a portion of the ansible-operator image, in particular the dependencies section. Now that image builds run on each PR, it would be expedient to have some base containing all the required python deps. This image should also be rebuildable on a whim by an admin, should the need arise.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        if [[ $GITHUB_REF == refs/tags/* ]]; then
+          TAG=${GITHUB_REF#refs/tags/}
+          MAJOR_MINOR=${TAG%.*}
+          echo ::set-output name=tags::${IMG}:${TAG},${IMG}:${MAJOR_MINOR}
+
+        elif [[ $GITHUB_REF == refs/heads/* ]]; then
+          TAG=$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
+          if [ "${{ github.event.repository.default_branch }}" = "$TAG" ]; then
+            TAG=edge
+          fi
```
Comment (joelanford) [inline .github/workflows/deploy.yml:75]: Consider removing this if you want to stick with branch based tag name, e.g. `quay.io/operator-framework/operator-sdk:master`

**Pull request closed without merging**